include/envoy/upstream/upstream.h

```diff
@@ -812,6 +826,11 @@ class ClusterInfo {
    */
   virtual ClusterLoadReportStats& loadReportStats() const PURE;
 
+  /**
+   * @return ClusterTimeoutBudgetStats& stats on timeout budgets for this cluster.
+   */
+  virtual ClusterTimeoutBudgetStats& timeoutBudgetStats() const PURE;
```

Comment:
If the feature is not being used, can we not allocate anything for it? So perhaps return an optional std::reference? If we do that we can:
1) Remove all the null histogram changes I think
2) Make the router code branch and avoid all of the timing code if we are not using this feature?